VirtoCommerce.ImageToolsModule.Data/ThumbnailGeneration/ImagesChangesProvider.cs

```diff
@@ -155,10 +157,15 @@ protected virtual ICollection<BlobInfo> GetOriginalItems(ICollection<BlobInfo> s
 
             foreach (var blobInfo in source)
             {
+                var name = blobInfo.FileName;
```

Comment:
I think the better place for this kind of filtration is `ReadBlobFolder` method. And better use `Path.GetExtension` method instead of manual  manipulation with a file name

Reply:
Agree, made an update, left the "name" variable outside the second iteration in "GetOriginalItems" to avoid excess operations.